### PR TITLE
Erikdstock/refactor/shipping2 naming

### PR DESCRIPTION
**Pull request by erikdstock**
The type of this PR is: **refactor**
This PR completes the refactor in #13297 by fixing tests and making a few more refactoring optimizations and simplifications.

### CODE REVIEW

File: src/Apps/Order/Routes/Shipping2/Components/FulfillmentDetailsForm.tsx [hunk 1]
```diff
@@ -117,11 +117,7 @@ const FulfillmentDetailsFormLayout = (
      * Pass some key formik bits up to the shipping route
      * TODO: This could be accomplished with useImperativeHandle(ref, formikContext)
      */
-    shippingContext.actions.setFormHelpers({
-      submitForm,
-      isValid: isValid,
-      values: values,
-    })
+    shippingContext.actions.setFormHelpers(formikContext)
```
Comment (damassi) [inline src/Apps/Order/Routes/Shipping2/Components/FulfillmentDetailsForm.tsx:120]: 👌

File: src/Apps/Order/Routes/Shipping2/Components/ShippingQuotes2.tsx [hunk 2]
```diff
@@ -43,6 +44,7 @@ export const ShippingQuotes2: React.FC<ShippingQuotesProps> = ({
   )
 
   const quotes = extractNodes(data.shippingQuoteOptions)
+  useAutoSelectBestShippingQuote(quotes)
```
Comment (damassi) [inline src/Apps/Order/Routes/Shipping2/Components/ShippingQuotes2.tsx:47]: linebreak

File: src/Apps/Order/Routes/Shipping2/Components/ShippingQuotes2.tsx [hunk 3]
```diff
@@ -126,3 +128,42 @@ export const shippingQuoteDisplayNames = {
...
   priority: "Priority",
 }
+
+const useAutoSelectBestShippingQuote = (
+  quotes: Array<{
+    readonly id: string
+    readonly isSelected: boolean
+    readonly price: string | null | undefined
+    readonly priceCents: number
+    readonly typeName: string
```
Comment (damassi) [inline src/Apps/Order/Routes/Shipping2/Components/ShippingQuotes2.tsx:138]: pull this from the associated relay type (always) -- never retype things that are auto-generated.

File: src/Apps/Order/Routes/Shipping2/Components/ShippingQuotes2.tsx [hunk 4]
```diff
@@ -126,3 +128,42 @@ export const shippingQuoteDisplayNames = {
...
+  }>
+) => {
+  const shippingContext = useShippingContext()
```
Comment (damassi) [inline src/Apps/Order/Routes/Shipping2/Components/ShippingQuotes2.tsx:141]: linebreak

File: src/Apps/Order/Routes/Shipping2/Components/ShippingQuotes2.tsx [hunk 5]
```diff
@@ -126,3 +128,42 @@ export const shippingQuoteDisplayNames = {
...
+  const bestArtsyShippingQuoteId =
+    quotes.find(
+      quote => quote.id === shippingContext.state.selectedShippingQuoteId
+    )?.id ||
+    quotes.find(quote => quote.isSelected)?.id ||
+    quotes?.[0]?.id
+
+  const setSelectedShippingQuote =
+    shippingContext.actions.setSelectedShippingQuote
+  useEffect(() => {
```
Comment (damassi) [inline src/Apps/Order/Routes/Shipping2/Components/ShippingQuotes2.tsx:154]: Line break between 153 and 154 (again, try really really hard to make this an automatic habit so I can stop flagging basic readability stuff in every PR 🙏 readability has historically been something that needs to improve, but you have to work with me; i'm not flagging this stuff to be pedantic)

File: src/Apps/Order/Routes/Shipping2/ShippingContext.tsx [hunk 6]
```diff
@@ -58,19 +58,19 @@ export const ShippingContextProvider: FC<Pick<
...
       // unchecks it after saving it in the fulfillment details step.
-      values: ({
+      values: {
         attributes: {
           saveAddress: false,
         },
-      } as unknown) as FulfillmentValues,
-    },
+      },
+    } as unknown) as FormHelpers,
```
Comment (damassi) [inline src/Apps/Order/Routes/Shipping2/ShippingContext.tsx:73]: 👍

**Review by damassi — APPROVED** [review]
Changes look good and thank you so much for finding / fixing that test. I knew it was something silly 👍 just a few (minor!) readability concerns that will continue to flag in order to encourage a shift towards better habits.

File: src/Apps/Order/Routes/Shipping2/Components/ShippingQuotes2.tsx (hunk 5 above)
Comment (damassi) [reply src/Apps/Order/Routes/Shipping2/Components/ShippingQuotes2.tsx:154]: To be clear, here are the ground rules: _things that are different from one another_ require a line break in between them. A variable is different from a useEffect; a code comment is different from an unrelated variable; a for loop is different from a function call -- and so on and so forth. Internalize this completely! Help your comrades out 👍

**Comment by erikdstock** [comment]
Thanks @damassi - responded to feedback and just went ahead and pushed onto the PR branch so this is closed now.

**Comment by damassi** [comment]
🙏 thanks again

**Pull request merged**